src/main/kotlin/com/example/ticketbookingapp/domain/Reservation.kt

```diff
@@ -0,0 +1,20 @@
+﻿package com.example.ticketbookingapp.domain
+
+import jakarta.persistence.*
+import jakarta.validation.Valid
+import java.time.Instant
+
+@Entity
+class Reservation(
+    @get:Valid
+    val user: User,
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    val movieScreening: MovieScreening,
+
+    @Column(nullable = false)
+    val expirationDate: Instant,
```

Comment:
Powiesz czemu `Instant` a nie np. `LocalDateTime`? Zdradzę, że mi akurat takie podejście się podoba, ale ciekaw jestem jakie Ty masz powody ;)

Reply:
Chciałem uniknąć problemów z strefami czasowymi.

Instant odpowiada pewnemu unix time (albo alternatywnie czasowi w utc), nie interesują go takie udziwnienia jak strefy czasowe.

Gdybym miał w memory obekty typu LocalDateTime to bym się musiał zastanawiać której strefie czasowej one odpowiadają. Wówczas bym przyjął, że odpowiada strefie czasowej utc, ale to wówczas bym równie dobrze (a właściwie lepiej) mógł używać Instant. A jakbym przyjął, że LocalDateTime odpowiadają strefie czasowej kraju w którym znajduje się kino to myślę, że bym otworzył puszkę Pandory, np związaną z handlowaniem zmian strefy czasowej, albo handlowaniem tego że część kin jest w jednym kraju a część w innym.

W settingsach mam linię
spring.jpa.properties.hibernate.jdbc.time_zone=UTC
by te Instanty mi się zapisywały w utc w bazie danych